src/js/common/schemaform/components/SchemaForm.jsx

```diff
@@ -136,6 +121,21 @@ class SchemaForm extends React.Component {
   }
 
   render() {
+    const fields = {
```

Comment:
Is it possible to modify the tests instead of moving these? This component is re-rendered a lot when you use a form.

Reply:
I was wrong when I said fields was modified. The Issue here is actually the way node/babel resolves circular dependencies. 

The random test order means that when the `ArrayField` spec is run before a spec that uses `SchemaForm`, `ArrayField` is imported and `SchemaForm` is imported somewhere down that load chain. Since `ArrayField` hasn't finished loading it is `undefined` at the time node evaluates the `SchemaForm` file when the `fields` and `reviewFields` objects are created.

Moving the object definitions inside `render` means they aren't evaluated until the object is instantiated, so `ArrayField` will have finished loading by that point. 

I don't think there's anyway around this that leaves these object defined only once outside the class body. They don't need to be in the render method and so could be somewhere else if you think it'd be more faster.

Reply:
Could we solve both problems (test failures and efficiency) by making these properties of the `SchemaForm` class?

Reply:
Yeah, I think that's probably the way to go. We can create these objects in the constructor, then. We're not creating lots of SchemaForm components usually, so I don't imagine that will have much impact.

Reply:
Sounds like a plan I updated to have them assigned as `this.fields` and `this.reviewFields` in the constructor.